src/components/networks/utils.js

```diff
@@ -142,59 +133,28 @@ export function validateIpv6(address) {
  * @returns {boolean}
  */
 export function validateIpv6Prefix(prefix) {
-    if (!/^[0-9]+$/.test(prefix.trim()))
-        return false;
-    const prefixLength = parseInt(prefix, 10);
-    if (isNaN(prefixLength) || prefixLength < 0 || prefixLength > 128)
-        return false;
-
-    return true;
-}
-
-/**
- * Converts ipv6 address to string containing it's binary representation
- *
- * @param {string} ip
- * @returns {string}
- */
-function ipv6ToBinStr(ip) {
-    const validGroupCount = 8;
-    /* Split address by `:`
-     * Then check if the array contains an empty string (happens at ::), and if so
-     * replace it with the appropriate number of 0 entries.
-     */
-    const arrAddr = ip.split(":");
-    const arrAddrExpanded = arrAddr.reduce((accum, hexNum) => {
-        if (hexNum)
-            accum.push(hexNum);
-        else
-            for (let i = 0; i < (validGroupCount - arrAddr.length + 1); i++)
-                accum.push("0");
-        return accum;
-    }, []);
-
-    /* Convert the array of 8 hex entries into a 128 bits binary string */
-    return arrAddrExpanded.map(num => {
-        let bin = parseInt(num, 16).toString(2);
-        while (bin.length < 16)
-            bin = "0" + bin;
-        return bin;
-    }).join("");
+    if (/^[0-9]+$/.test(prefix.trim())) {
+        try {
+            ipaddr.IPv6.subnetMaskFromPrefixLength(prefix);
+            return true;
+        } catch (_ex) {}
+    }
+    return false;
 }
 
 /**
- * Checks whetever IPv6 address @ip is in subnet defined by @network and @prefix
+ * Checks whetever IPv6 @address is in subnet defined by @network and @prefix
  *
  * @param {string} network
  * @param {string} prefix
- * @param {string} ip
+ * @param {string} address
  * @returns {boolean}
  */
-export function isIpv6InNetwork(network, prefix, ip) {
-    network = ipv6ToBinStr(network);
-    network = network.substring(0, prefix);
-    ip = ipv6ToBinStr(ip);
-    ip = ip.substring(0, prefix);
+export function isIpv6InNetwork(network, prefix, address) {
+    if (!validateIpv6(network) || !validateIpv6Prefix(prefix) || !validateIpv6(address))
+        return false;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1455-20240222-112249-96a33f93-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)